app/controllers/admin/variant_overrides_controller.rb

```diff
@@ -63,11 +63,12 @@ def inventory_import_dates
       options = [{ id: '0', name: 'All' }]
 
       import_dates = VariantOverride.
-        select('variant_overrides.import_date').
+        select(:import_date).
         where('variant_overrides.hub_id IN (?)
-        AND variant_overrides.import_date IS NOT NULL', editable_enterprises.collect(&:id))
+        AND variant_overrides.import_date IS NOT NULL', editable_enterprises.collect(&:id)).
+        order('import_date DESC')
 
-      import_dates.uniq.collect(&:import_date).sort.reverse.map { |i| options.push(id: i.to_date, name: i.to_date.to_formatted_s(:long)) }
+      import_dates.collect(&:import_date).uniq.map { |i| options.push(id: i.to_date, name: i.to_date.to_formatted_s(:long)) }
```

Comment:
My point is that we need to fetch data in a declarative way, embracing SQL (sounds like a pretty cool sales pitch :trollface: ). That is, let's not think about the control flow but about what data we want to get from the DB. It'll get that data for us. I can recommend http://patshaughnessy.net/2015/6/18/dont-let-your-data-out-of-the-database as perfect article on that matter.

To sum up, what about:

```ruby
SELECT DISTINCT(import_date)
FROM varaint_overrides
INNER JOIN enterprises ON enterprises.id = hub_id
WHERE <constraints to get editable_enterprises>
AND import_date IS NOT NULL
ORDER BY import_date DESC
```

which would be written in ActiveRecord as:

```ruby
VariantOverride
  .select(:import_date)
  .distinct
  .join(:enterprises)
  .merge(<constraints to get editable_enterprises>)
  .where('import_date IS NOT NULL')
  .order(import_date: :desc)
```

What I see after digging a little bit is that `<constraints to get editable_enterprises>` might be a bit hard to get as a series of `WHERE` constraints we can reuse since `editable_enterprises` is already a result set. Give it a try, though.

Reply:
When I tried to add `distinct` into the chain it kept throwing errors, but it seems to work with `.select('DISTINCT variant_overrides.import_date')`, which is how distinct queries are done elsewhere in the codebase.